Feature/subscription email message

MartynasKasp opened the pull request.

MykolasVitkus committed:
Merge pull request #22 from MykolasVitkus/dev-master

Dev master

MykolasVitkus committed:
link to user in comments

MykolasVitkus committed:
changed to 3 comments per event page

MykolasVitkus committed:
comments sorted by date(newest first)

MartynasKasp committed:
EmailManager service class created

MartynasKasp committed:
Fixed translation to lithuanian

MartynasKasp committed:
Fixed translation

MartynasKasp committed:
Email message is being sent to users, who are subscribed to new event's category.

MykolasVitkus commented on src/Controller/PasswordResetController.php:
```diff
@@ -8,20 +8,21 @@
      * @Route("/password_reset", name="password_reset")
      */
-    public function index(Request $request, \Swift_Mailer $mailer)
+    public function index(Request $request, \Swift_Mailer $mailer, EmailManager $emailManager)
```
O cia kur dar naudoji swift maileri? Nematau kur

MartynasKasp replied:
54 failo eilutė
` $mailer->send($emailMessage);`

MartynasKasp replied:
> O cia kur dar naudoji swift maileri? Nematau kur @MykolasVitkus 

54 failo eilutė
`$mailer->send($emailMessage);`

MykolasVitkus replied:
Ai, tai jeigu email manageri ji ir issiustum ta maila? Nereiktu controllery injectint swiftmailerio, kanagi jis already jau yra emailManager

MartynasKasp replied:
> O cia kur dar naudoji swift maileri? Nematau kur

54 failo eilutė
`$mailer->send($emailMessage);`

MykolasVitkus commented on src/Service/EmailManager.php:
```diff
@@ -0,0 +1,31 @@
+    public function sendEmail(string $messageTitle, string $sendToEmail, string $template, string $textType, array $variables = []): \Swift_Message
+    {
+        $messageToSend = (new \Swift_Message($messageTitle))
+            ->setFrom('bemantelio@gmail.com')
```
Gerai butu owner email i env isidet

pockeviciuspovilas reviewed (approved):
Nice job!! 😁😁

MykolasVitkus commented on src/Controller/MailConfirmController.php:
```diff
@@ -7,53 +7,52 @@
      * @Route("/confirm", name="confirm_email")
      */
-    public function index(Request $request, \Swift_Mailer $mailer)
+    public function index(Request $request, \Swift_Mailer $mailer, EmailManager $emailManager)
```
Cia irgi swiftmailerio nebenaudoji lygtais

MykolasVitkus commented on src/Controller/EventController.php:
```diff
@@ -252,4 +258,34 @@ private function generateFilterQuery($formData, $queryBuilder)
         }
     }
+
+    private function sendSubscriptionEmail($event, \Swift_Mailer $mailer, EmailManager $emailManager)
```
Ir cia nenaudoji:D

MykolasVitkus commented on templates/events/view.html.twig:
```diff
@@ -77,7 +77,7 @@
                             <div class="col-md-9">
                                 <p class="float-left">
-                                    {{ comment.getAuthor().getEmail()}}
+                                    <a href="{{ path('app_userProfile', { 'id': comment.author.id }) }}">{{ comment.getAuthor().getEmail()}}</a>
```
O sita as pamirsau supushint?:D nes lygtais buvau padares

MartynasKasp replied:
Tai matyt. Ar tu į dev-master pushinai šitus? Nes aš man atrodo iš master brancho šį kūriau

MykolasVitkus commented on templates/security/login.html.twig:
```diff
@@ -22,7 +22,7 @@
             </div>
             <input type="checkbox" id="remember_me" name="_remember_me" checked />
-            <label for="remember_me">Palikite mane prisijungųsį</label>
+            <label for="remember_me">Palikite mane prisijungusį</label>
```
Cia mano gramatines klaidas taisai, ane? :D im insulted

MartynasKasp replied:
O aš triggered tokių klaidų :D

MartynasKasp committed:
Removed unnecessary usage of Swift_Mailer argument in functions.
Swift_Mailer message is now being sent from EmailManager service.
Email address added to dotenv.

MykolasVitkus reviewed (approved):
Good, next time use flash messages;)

MartynasKasp committed:
Using flash messages instead of variables for success/error messages.

MartynasKasp committed:
Avatar will be now set for new created admin users

MartynasKasp committed:
Fix for flash message showing

DreDasLT commented:
Gerai padirbėta :)

DreDasLT reviewed (approved):
Gerai padirbėta :)

MartynasKasp committed:
style edited
